src/core/service.c

```diff
@@ -423,7 +423,7 @@ static int on_fd_store_io(sd_event_source *e, int fd, uint32_t revents, void *us
         return 0;
 }
 
-static int service_add_fd_store(Service *s, int fd, const char *name) {
+static int service_add_fd_store(Service *s, int fd, const char *name, bool poll) {
```

Comment:
hmm, i can't say I like that the parameter is named exactly like the function. that's a namespace clash in my book. Just call this do_poll or something like that, to avoid any ambiguities.